### ISSUE TITLE
Regression in `prop-types` rule v7.13.0

### ISSUE THREAD

**Issue by willdurand** [yannickcr/eslint-plugin-react#2298]
The `prop-types` rule reports false-positives when we upgraded from version `7.12.0` to `7.13.0`. I believe this is related to #2241 and Flow.

## Example

```js
type Props = {|
  _value?: number
|};

function App({ _value = defaultValue }: Props = {}) {
  ...
}
```

### What happened?

```
error  '_value' is missing in props validation  react/prop-types
```

### What did you expect to happen?

No error.

## Other information

It worked until `7.12.0` but it fails with `7.13.0`.

I created a demo project to reproduce the bug: https://github.com/willdurand/eslint-issue-prop-types.

**Comment by ljharb** [comment]
I think the issue is that you've defaulted the props arg to `{}`, for no reason I can see - props are unconditionally provided to react components.

What happens if you remove `= {}`?

**Comment by yannickcr** [comment]
Thank you for the demo project, it was very helpful.

It should be fixed in next release 🙂
